source/Halibut/ServiceModel/RequestCancellationTokens.cs

```diff
@@ -11,28 +11,37 @@ public RequestCancellationTokens(CancellationToken connectingCancellationToken,
         {
             ConnectingCancellationToken = connectingCancellationToken;
             InProgressRequestCancellationToken = inProgressRequestCancellationToken;
-        }
-
-        public CancellationToken ConnectingCancellationToken { get; set; }
-        public CancellationToken InProgressRequestCancellationToken { get; set; }
 
-        public CancellationToken LinkedCancellationToken => LazyLinkedCancellationToken.Value;
-
-        Lazy<CancellationToken> LazyLinkedCancellationToken => new (() =>
-        {
             if (ConnectingCancellationToken == CancellationToken.None && InProgressRequestCancellationToken == CancellationToken.None)
             {
-                return CancellationToken.None;
+                LinkedCancellationToken = CancellationToken.None;
+            }
+            else if (InProgressRequestCancellationToken == CancellationToken.None)
+            {
+                LinkedCancellationToken = ConnectingCancellationToken;
             }
+            else if (ConnectingCancellationToken == CancellationToken.None)
+            {
+                LinkedCancellationToken = InProgressRequestCancellationToken;
+            }
+            else
```

Comment:
Does another optimisation exists where if the `ConnectingCancellationToken` and `InProgressRequestCancellationToken` are reference equal then use either as the linked token

Reply:
Yes good call. It's not a code path that would be triggered in our current usage but good to optimise for future usage 👍